react-app-drawer-component/my-app/src/AppDrawer.js

```diff
@@ -9,31 +9,44 @@ import './AppDrawer.css';
  * @param {items} An array containing pages listed within the menu.
  */
 
-export default function AppDrawer({heading, items, onTitleChange, onMenuToggle, view}) {
+export default function AppDrawer({heading, items, onTitleChange}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const overlay = (!isOpen) ?
+    { backgroundColor: 'rgba(255, 255, 255, 0)', pointerEvents: 'none' } :
+    { backgroundColor: 'rgba(255, 255, 255, 0.25)', pointerEvents: 'all' };
+  function toggleMenu() {
+    setIsOpen(!isOpen);
+  }
 
   return (
-    <div style={{position: 'absolute'}}>
-      {(view === 'main') && <FaHamburger className="hamburger-icon" onClick={onMenuToggle}/>}
-      <DrawerItems heading={heading}
-        items={items}
-        view={view}
-        onMenuToggle={onMenuToggle}
-        onTitleChange={onTitleChange}
-      />
-    </div>
+    <>
+      <div style={{position: 'absolute'}}>
+        {(!isOpen) && <FaHamburger className="hamburger-icon" onClick={toggleMenu}/>}
+        <DrawerItems heading={heading}
+          items={items}
+          isOpen={isOpen}
+          onMenuToggle={toggleMenu}
+          onTitleChange={onTitleChange}
+        />
+      </div>
+      <div style={overlay}
+          className="overlay"
+          onClick={() => isOpen && toggleMenu()}>
```

Comment:
In theory, this click will never happen if `isOpen` is false, but your short-circuit check is a nice safety feature.

It might be possible to instead only render the overlay div if `isOpen` is true:
`{isOpen && <div style={overlaly} ...>`
The standard solution doesn't do that, so I'm not sure it would work, but it seems like it would. Then you wouldn't need to have two different styles for the overlay. It seems that would simplify the code even further.